evap/results/tools.py

```diff
@@ -215,48 +215,42 @@ def calculate_average_grade_and_distribution(course):
             # collect the grade question average, non-grade question average and the distributions for each contributor
             contributor_averages.append(mix(grade_questions_average, non_grade_questions_average, settings.CONTRIBUTOR_GRADE_QUESTIONS_PERCENTAGE))
-            grade_questions_distribution = normalize_distribution(grade_questions_counts)
-            non_grade_questions_distribution = normalize_distribution(non_grade_questions_counts)
+            grade_questions_distribution = grade_questions_distribution
+            non_grade_questions_distribution = non_grade_questions_distribution
             contributor_distributions.append(mix_distributions(grade_questions_distribution, non_grade_questions_distribution, settings.CONTRIBUTOR_GRADE_QUESTIONS_PERCENTAGE))
         else:
             # calculate the grade question average, non-grade question average and the distributions for the course
             course_grade_questions_average = grade_questions_average
-            course_grade_questions_distribution = normalize_distribution(grade_questions_counts)
+            course_grade_questions_distribution = grade_questions_distribution
             course_non_grade_questions_average = non_grade_questions_average
-            course_non_grade_questions_distribution = normalize_distribution(non_grade_questions_counts)
+            course_non_grade_questions_distribution = non_grade_questions_distribution
 
     # calculate the average grade and distribution over all contributors
     contributors_average = avg(contributor_averages)
-    contributors_distribution = {1: 0, 2: 0, 3: 0, 4: 0, 5: 0}
-    for distribution in contributor_distributions:
-        for answer, value in distribution.items():
-            contributors_distribution[answer] += value
-    contributors_distribution = normalize_distribution(contributors_distribution)
+    contributors_distribution = avg_distribution(contributor_distributions)
 
     # calculate to which percentage each group influences the total grade
     contributions_percentage = settings.CONTRIBUTIONS_PERCENTAGE if contributors_average is not None else 0
     course_grade_questions_percentage = settings.COURSE_GRADE_QUESTIONS_PERCENTAGE if course_grade_questions_average is not None else 0
     course_non_grade_questions_percentage = 1 - settings.CONTRIBUTIONS_PERCENTAGE - settings.COURSE_GRADE_QUESTIONS_PERCENTAGE if course_non_grade_questions_average is not None else 0
-    if contributors_average or course_grade_questions_average or course_non_grade_questions_average:
-        scale = 1 / sum([contributions_percentage, course_grade_questions_percentage, course_non_grade_questions_percentage])
-        contributions_percentage *= scale
-        course_grade_questions_percentage *= scale
-        course_non_grade_questions_percentage *= scale
+    if not contributors_average and not course_grade_questions_average and not course_non_grade_questions_average:
+        return None, None
```

Comment:
this can be done earlier, maybe before the `# calculate the average grade` comment? also, [optional] please check whether something like `if not any(average for ..., average, .... in grouped_results.values()): return None, None` can be done right at the beginning of this function and if that looks better to you.

Reply:
check